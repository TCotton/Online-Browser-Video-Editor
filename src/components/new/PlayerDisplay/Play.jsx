import React from "react";
import {useDispatch} from "react-redux";
import {playFn} from '../slices/playerSlice';

export const Play = () => {
    const dispatch = useDispatch();
    const onClick = () => {
        dispatch(playFn(true))
    }
    return (
        <img src={"./play.svg"} className="play" alt="play" onClick={onClick}/>
    )
}