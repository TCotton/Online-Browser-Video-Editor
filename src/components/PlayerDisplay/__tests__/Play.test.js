import React from 'react'
import { render } from '@testing-library/react'
import { Play } from '../Play'

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: jest.fn()
}))

describe('Play', () => {
  it('should render successfully', () => {
    const { container } = render(<Play />)
    expect(container.querySelector('[data-testid="play"]')).not.toBeNull()
  })

  it('should match snapshot', () => {
    const { baseElement } = render(<Play />)
    expect(baseElement).toMatchSnapshot()
  })
})
