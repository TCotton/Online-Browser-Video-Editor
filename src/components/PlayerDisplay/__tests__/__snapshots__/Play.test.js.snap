// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Play should match snapshot 1`] = `
<body>
  <div>
    <img
      alt="play"
      class="play"
      data-testid="play"
      src="./play.svg"
    />
  </div>
</body>
`;
